lib/max_subarray.py

```diff
@@ -2,11 +2,17 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
+        
     """
     if nums == None:
         return 0
     if len(nums) == 0:
         return 0
-    pass
+    maxSub = nums[0]
+    currentSum = nums[0]
+    for index in range(1, len(nums)):
+        currentSum = max(nums[index], currentSum + nums[index])
```

Comment:
✨ This is a really nice way to represent this calculation, which captures the underlying invariant that makes Kadane's algorithm work.